src/jit/jiteh.cpp

```diff
@@ -432,6 +432,29 @@ bool                Compiler::bbInTryRegions(unsigned regionIndex, BasicBlock *
 }
 
 /*
+Given a block and an eh region index, check to see if an exception raised
+in the block could be handled by the given region (possibly after inner
+regions).  For this check, a funclet is considered to be in the region
+it was extracted from.
+*/
+bool                Compiler::bbInExnFlowRegions(unsigned regionIndex, BasicBlock * blk)
+{
+  assert(regionIndex < EHblkDsc::NO_ENCLOSING_INDEX);
+  EHblkDsc* ExnFlowRegion = ehGetBlockExnFlowDsc(blk);
+  unsigned tryIndex = (ExnFlowRegion == nullptr ? EHblkDsc::NO_ENCLOSING_INDEX : ehGetIndex(ExnFlowRegion));
+
+  // Loop outward until we find an enclosing try that is the same as the one
+  // we are looking for or an outer/later one
+  while (tryIndex < regionIndex)
+  {
+    tryIndex = ehGetEnclosingTryIndex(tryIndex);
+  }
+
+  // Now we have the index of 2 try bodies, either they match or not!
+  return (tryIndex == regionIndex);
+}
+
```

Comment:
btw, this seems to be true throughout your new code.

---
In reply to: [61190340](https://github.com/dotnet/coreclr/pull/4591#discussion_r61190340) [](ancestors = 61190340)